billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/builders/impl/PTManualCreditNoteEntryBuilderImpl.java

```diff
@@ -80,22 +80,22 @@ protected void validateInstance() throws BillyValidationException {
         PTCreditNoteEntryEntity cn = this.getTypeInstance();
         // The <generic> specs below are necessary because type inference fails here for unknown reasons
         // If removed, these lines will fail in runtime with a linkage error (ClassCastException)
-        BillyValidator.<BigDecimal>mandatory(cn.getQuantity(),
-                PTGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.quantity"));
-        BillyValidator.<String>mandatory(cn.getUnitOfMeasure(),
-                PTGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.unit"));
-        BillyValidator.<Product>mandatory(cn.getProduct(),
+        BillyValidator.mandatory(cn.getQuantity(),
+								 PTGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.quantity"));
+        BillyValidator.mandatory(cn.getUnitOfMeasure(),
+								 PTGenericInvoiceEntryBuilderImpl.LOCALIZER.getString("field.unit"));
+        BillyValidator.<Object>mandatory(cn.getProduct(),
```

Comment:
Porque é que aqui precisa de `<Object>`?

Reply:
(copy) Isto vem na sequência de uma correcções que o froque fez para passar de java 7 para o 8. Pelos visto o type inference está diferente e ele estava a rebentar com o cast de Product para CharSequence porque estava a inferir o mandatory incorrecto.
O froque mandou-me esta questão do stackoverflow que parece dizer respeito ao mesmo problema.
https://stackoverflow.com/questions/59430039/why-does-the-compiler-choose-this-generic-method-with-a-class-type-parameter-whe